src/js/actions/sampler.js

```diff
@@ -554,15 +551,23 @@ define(function (require, exports) {
                 textAlignmentPromise,
                 textStylePromise,
                 effectsPromise)
+            .bind(this)
             .then(function () {
                 return descriptor.endTransaction(transaction);
+            })
+            .then(function () {
+                return this.transfer(layerActions.resetLayers, document, targetLayers);
```

Comment:
Added these, because during hte transactions, our resetLayers / resetBorderPolicies do nothing (as the local data structures don't get updated)